src/bases.d.ts

```diff
@@ -58,7 +58,7 @@ export interface Evented extends Destroyable {
 	 * @param listener A listener or array of listeners that accept error events
 	 * @returns A handle which can be used to remove the listeners
 	 */
-	on(type: 'error', listener: EventedListenerOrArray<this, EventErrorObject<this>>): Handle;
+	on<T>(type: 'error', listener: EventedListenerOrArray<T, EventErrorObject<T>>): Handle;
```

Comment:
It will always be an object that extends `Stateful` but don't see much harm in it being `T`, would this allow the type to be narrowed even further?